src/blocks.zig

```diff
@@ -414,23 +414,45 @@ pub const Block = packed struct { // MARK: Block
 
 // MARK: Tick
 pub var tickFunctions: utils.NamedCallbacks(TickFunctions, TickFunction) = undefined;
-pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void;
+pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32, _param: ?*anyopaque) void;
 pub const TickFunctions = struct {
-	pub fn replaceWithCobble(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void {
-		std.log.debug("Replace with cobblestone at ({d},{d},{d})", .{x, y, z});
-		const cobblestone = parseBlock("cubyz:cobblestone");
+	pub const FunctionName = enum {
+		null,
+		replaceWith,
+	};
+
+	inline fn castToArgs(comptime Args: type, param: *anyopaque) *Args {
+		return @ptrCast(@alignCast(param));
+	}
+
+	const ReplaceWithArgs = struct {
+		block: []const u8,
```

Comment:
Should parse the block during load instead of reparsing it at runtime for every tick event.